Typist/ViewController.swift

```diff
@@ -27,15 +27,14 @@ class ViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        // TODO: add keyboard toolbar handler
+        // keyboard input accessory view support
         textField.inputAccessoryView = UIView(frame: toolbar.bounds)
         keyboard.scrollView = tableView
         
-        // starts frame observer
+        // keyboard frame observer
         keyboard.on(event: .willChangeFrame) { [unowned self] options in
             let height = UIScreen.main.bounds.height - options.endFrame.origin.y
             self.bottom.constant = max(0, height - self.toolbar.bounds.height)
-//            UIView.animate(withDuration: options.animationDuration, delay: 0, options: UIViewAnimationOptions(curve: options.animationCurve), animations: {
             UIView.animate(withDuration: 0) {
                 self.tableView.contentInset.bottom = max(self.toolbar.bounds.height, height)
```

Comment:
Maybe we could include all of those recalculated values in `options` if possible? Another idea—pass toolbar configuration to keyboard and options will provide all the geometry pre-recalculated as `CGRect`?

Reply:
The main issue revolves around the toolbar height being included in the keyboard height. We need to figure out how to make this easier to handle.